app/src/main/java/chat/rocket/android/contacts/adapter/ContactViewHolder.kt

```diff
@@ -0,0 +1,48 @@
+package chat.rocket.android.contacts.adapter
+
+import android.content.res.Resources
+import android.view.View
+import androidx.core.view.isGone
+import androidx.core.view.isVisible
+import chat.rocket.android.chatrooms.adapter.ViewHolder
+import kotlinx.android.synthetic.main.item_contact.view.*
+
+class ContactViewHolder(itemView: View) : ViewHolder<ContactItemHolder>(itemView) {
+
+    private val resources: Resources = itemView.resources
+
+
+    override fun bindViews(data: ContactItemHolder) {
+        val contact = data.data
+        with(itemView) {
+            contact_image_avatar.setImageURI(contact.getAvatarUrl())
+            contact_name.text = contact.getName()
+
+            if (contact.getUsername()==null) {
+                invite_contact.isVisible = true
+                chat_username.isGone=true
+            }else{
+                invite_contact.isGone= true
+                chat_username.isVisible=true
+                chat_username.text= "@"+contact!!.getUsername()
+            }
+
+            if (contact!!.isPhone()) {
+                contact_detail.text = contact!!.getPhoneNumber()
+            } else {
+                contact_detail.text = contact!!.getEmailAddress()
+            }
```

Comment:
This whole block of code where you set the phone number or the email detail is only relevant to the "if" statement above, when contact.getUsername() == null, correct?  Can you please move this block under that case please?

Reply:
No it cannot be in one case since getusername() depends on contacts that are not on RC and phone number depends on detail the user have mainly for contacts that are not on RC

Reply:
```if (contact.getUsername()==null) {
            invite_contact.isVisible = true
            chat_username.isGone=true

            if (contact!!.isPhone()) {
                contact_detail.text = contact!!.getPhoneNumber()
            } else {
                contact_detail.text = contact!!.getEmailAddress()
            }
    } else {
            invite_contact.isGone= true
            chat_username.isVisible=true
            chat_username.text= "@"+contact!!.getUsername()
     }
```

Does this work?